### DISCUSSION TITLE
Anyone knows why Mayastor io-engine is using magic number 512 to check if there are enough huge pages?

### DISCUSSION THREAD

**Discussion by cmontemuino** [openebs/openebs#3762] [Q&A] [answered]
This is the specific line used for checking if there are enough huge pages:

```rust
    if nr_pages + nr_1g_pages * 512 < PAGES_NEEDED {
    # ...
```
> https://github.com/openebs/mayastor/blob/v2.7.0/io-engine/src/bin/io-engine.rs#L42

According Mayastor's [installation guide](https://openebs.io/docs/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-installation#general), it requires **a minimum of 2GiB of 2MiB-sized pages**.

If my math is correct, the requirement is to have 1024 pages, which is exactly what's configured in:

```rust
const PAGES_NEEDED: u32 = 1024;
```
> https://github.com/openebs/mayastor/blob/v2.7.0/io-engine/src/bin/io-engine.rs#L42

I cannot figure out expression `nr_1g_pages * 512`. 512 pages with size 1GiB is 512GiB, right?

Am I missing something?

**Comment by tiagolobocastro** [comment, marked as answer]
Probably not the clearest piece of code.
1page of 1GiB pages is equivalent to 512 pages of 2MiB pages

**Reply by cmontemuino** [reply]
Got it!!! Somehow I was mixing page sizes. We're always talking about of 2MiB page size.

Perhaps a comment would help, but still the code is clear enough.

Thanks!!
